update to base16-bytestring-1.0

We are blocked on dependencies (https://github.com/haskell-hvr/resolv/pull/10, ...), but we need code changes as well

```
[ 16 of 180] Compiling Distribution.Client.HashValue ( src/Distribution/Client/HashValue.hs, /code/shared-haskell/cabal/dist-newstyle/build/x86_64-linux/ghc-8.6.5/cabal-install-3.5.0.0/x/cabal/build/cabal/cabal-tmp/Distribution/Client/HashValue.o ) [Data.ByteString.Base16 changed]

src/Distribution/Client/HashValue.hs:74:7: error:
    • Couldn't match expected type ‘Either String BS.ByteString’
                  with actual type ‘(BS.ByteString, BS.ByteString)’
```

--- a/cabal-install/cabal-install.cabal.dev
+++ b/cabal-install/cabal-install.cabal.dev
@@ -212,7 +212,7 @@ library cabal-lib-client
         async      >= 2.0      && < 2.3,
         array      >= 0.4      && < 0.6,
         base       >= 4.8      && < 4.15,
-        base16-bytestring >= 0.1.1 && < 0.2,
+        base16-bytestring >= 0.1.1 && < 1.1.0.0,
         binary     >= 0.7.3    && < 0.9,
         bytestring >= 0.10.6.0 && < 0.11,
         Cabal      == 3.5.*,

--- a/cabal-install/cabal-install.cabal.prod
+++ b/cabal-install/cabal-install.cabal.prod
@@ -267,7 +267,7 @@ executable cabal
         async      >= 2.0      && < 2.3,
         array      >= 0.4      && < 0.6,
         base       >= 4.8      && < 4.15,
-        base16-bytestring >= 0.1.1 && < 0.2,
+        base16-bytestring >= 0.1.1 && < 1.1.0.0,
         binary     >= 0.7.3    && < 0.9,
         bytestring >= 0.10.6.0 && < 0.11,
         Cabal      == 3.5.*,

--- a/cabal-install/cabal-install.cabal.zinza
+++ b/cabal-install/cabal-install.cabal.zinza
@@ -16,7 +16,7 @@ Version:            3.5.0.0
         async      >= 2.0      && < 2.3,
         array      >= 0.4      && < 0.6,
         base       >= 4.8      && < 4.15,
-        base16-bytestring >= 0.1.1 && < 0.2,
+        base16-bytestring >= 0.1.1 && < 1.1.0.0,
         binary     >= 0.7.3    && < 0.9,
         bytestring >= 0.10.6.0 && < 0.11,
         Cabal      == 3.5.*,

